tests/functional/test_accounts.py

```diff
@@ -81,15 +92,46 @@ def test_transfer_without_value_send_everything_false(sender, receiver):
         sender.transfer(receiver, send_everything=False)
 
 
-def test_transfer_without_value_send_everything_true(sender, receiver, isolation):
+def test_transfer_without_value_send_everything_true(sender, receiver):
+    initial_receiver_balance = receiver.balance
+    initial_sender_balance = sender.balance
+
     # Clear balance of sender
-    sender.transfer(receiver, send_everything=True)
+    receipt = sender.transfer(receiver, send_everything=True)
+
+    value_given = receipt.value
+    total_spent = value_given + receipt.total_fees_paid
+    assert sender.balance < 3000000000000  # Part of gas not spent remains
+    assert sender.balance == initial_sender_balance - total_spent
+    assert receiver.balance == initial_receiver_balance + value_given
 
     expected_err_regex = r"Sender does not have enough to cover transaction value and gas: \d*"
     with pytest.raises(AccountsError, match=expected_err_regex):
         sender.transfer(receiver, send_everything=True)
 
 
+def test_transfer_without_value_send_everything_true_with_gas_specified(
+    sender, receiver, eth_tester_provider
+):
+    initial_receiver_balance = receiver.balance
+    initial_sender_balance = sender.balance
+
+    # The gas selected here is very high compared to what actually gets used.
+    gas = 25000000
```

Comment:
if you change this to `2100`, it will have the same effect as the test above and the second transfer will fail

Reply:
good find!